### PR TITLE
One App v6

### PR DESCRIPTION
**Pull request by 10xLaCroixDrinker**
## Description

- Drop Node 12 support
- Update to react 17

## Motivation and Context

Node 12 is reaching EOL and desparately need to update react

## How Has This Been Tested?

`npm test`

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [x] Dependency update
- [ ] Security update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

They'll need to update node locally. Modules will require `one-amex.compatibility` in `package.json` to be updated. Modules should update themselves to react 17, but if they're loaded in this version of one-app they'll use react@17 either way

### CODE REVIEW

File: __tests__/client/initClient.spec.js [hunk 1]
```diff
@@ -119,33 +108,8 @@ describe('initClient', () => {
     expect(promiseResolveSpy).toHaveBeenCalled();
   });
 
-  it('should return rejected promise if there is an error', async () => {
```
Comment (10xLaCroixDrinker) [inline __tests__/client/initClient.spec.js:122]: This spec was wrong

File: __tests__/client/initClient.spec.js [hunk 2]
```diff
@@ -166,35 +130,10 @@ describe('initClient', () => {
     expect(promiseResolveSpy).toHaveBeenCalled();
   });
 
-  it('should use strict mode', async () => {
```
Comment (10xLaCroixDrinker) [inline __tests__/client/initClient.spec.js:169]: This spec was also wrong

File: src/client/initClient.jsx [hunk 3]
```diff
@@ -60,6 +60,7 @@ export default async function initClient() {
     });
 
     /* eslint-disable react/jsx-props-no-spreading */
+    /* istanbul ignore next */
```
Comment (10xLaCroixDrinker) [inline src/client/initClient.jsx:63]: No purposeful need to actually render this in tests

File: __tests__/client/initClient.spec.js [hunk 4]
```diff
@@ -84,20 +75,18 @@ describe('initClient', () => {
...
-    loadPrerenderScripts.mockReturnValue(Promise.reject(mockError));
+    loadPrerenderScripts.mockImplementationOnce(() => { throw mockError; });
 
     const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
     const initClient = require('../../src/client/initClient').default;
 
-    try {
-      await initClient();
-    } catch (error) {
-      expect(consoleErrorSpy).toHaveBeenCalledWith(mockError);
```
Comment (10xLaCroixDrinker) [inline __tests__/client/initClient.spec.js:95]: This did not work

**Review by infoxicator — COMMENTED** [review]
with this now being a breaking change, are we able to remove `make-promises-safe` as discussed?

https://github.com/americanexpress/one-app/blob/main/package.json#L105

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Review by code-forger — APPROVED** [review]

**Pull request merged**